src/datetime.rs

```diff
@@ -33,13 +32,70 @@ pub struct DateTime {
 
 impl fmt::Display for DateTime {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}T{}", self.date, self.time)?;
+        //write!(f, "{}T{}", self.date, self.time)?;
+        //this change 129ns/iter => 52ns/iter
+        if self.time.microsecond != 0 {
+            let mut buf: [u8; 26] = *b"0000-00-00T00:00:00.000000";
+            buf[0] = b'0' + (self.date.year / 1000) as u8;
+            buf[1] = b'0' + (self.date.year / 100 % 10) as u8;
+            buf[2] = b'0' + (self.date.year / 10 % 10) as u8;
+            buf[3] = b'0' + (self.date.year % 10) as u8;
+            buf[5] = b'0' + (self.date.month / 10) as u8;
+            buf[6] = b'0' + (self.date.month % 10) as u8;
+            buf[8] = b'0' + (self.date.day / 10) as u8;
+            buf[9] = b'0' + (self.date.day % 10) as u8;
+            buf[11] = b'0' + (self.time.hour / 10) as u8;
+            buf[12] = b'0' + (self.time.hour % 10) as u8;
+            buf[14] = b'0' + (self.time.minute / 10) as u8;
+            buf[15] = b'0' + (self.time.minute % 10) as u8;
+            buf[17] = b'0' + (self.time.second / 10) as u8;
+            buf[18] = b'0' + (self.time.second % 10) as u8;
+            buf[19] = b'.';
```

Comment:
you can surely remove this?

Reply:
I can cut out some boilerplate code.But complete removal affects performance，Because buffer is pre-allocated